dashboard/app/models/contact_rollups_raw.rb

```diff
@@ -33,6 +88,36 @@ def self.extract_parent_emails
     ContactRollupsV2.execute_query_in_transaction(query)
   end
 
+  def self.extract_scripts_taught
+    source_sql = <<~SQL
+      SELECT email, scripts.name, MAX(se.updated_at) AS updated_at
+      FROM users u
```

Comment:
We should call out that this query only selects teachers in `users` table because `email > ''` condition implicitly rules out student accounts (whose emails are empty). 

Even better, we can reduce the cost of the JOIN by using `SELECT FROM users WHERE email > ''` as the left side of the JOIN instead of using the full `users` table. I can create a method for that subquery since we have to use it many times.

Reply:
Created `teacher_query` method in this commit for https://github.com/code-dot-org/code-dot-org/pull/35295/commits/52d4af3e56feb9fdc1ed45be7fe57b96228740f7 for https://github.com/code-dot-org/code-dot-org/pull/35295.

Reply:
I've never really understood query optimization...I feel like my expectation is that MySQL would be smart enough to filter out all of the rows without email addresses (ie, student accounts) before joining? Is that incorrect?

Reply:
Each db engine has its own way to optimize queries using multiple things such as indexes and column value distribution. In this case, the `users` table doesn't have index on `email`, so I don't know if the db will execute row filtering first or joining first.

The rule of thumbs for optimizing JOIN is to reduce the size of tables being joined, especially the table on the left side of the JOIN. This has to do with data living in memory or disk. If one of the tables (or both of them) is big and does not fit into the memory, the db will have to swap parts of the table in and out between disk and memory. To reduce number of swapping, we want to keep the smaller table in memory, while swapping the big table. We help the db engine by putting the smaller table on the left side of the JOIN.

Reply:
Thanks -- I can update to use your new method in a subsequent PR (our work isn't done here...)